### PR TITLE
fix(cmd): fix help string when git remote already exists

### PR DESCRIPTION
**Pull request by bacongobbler**
TODO:

 - [x] add tests

### CODE REVIEW

File: cmd/apps.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func (d DeisCmd) AppCreate(id, buildpack, remote string, noRemote bool) error {
 
 	if !noRemote {
 		if err = git.CreateRemote(s.Client.ControllerURL.Host, remote, app.ID); err != nil {
-			if err.Error() == "exit status 128" {
+			if strings.Contains(err.Error(), "Error when running 'git remote add") {
```
Comment (Joshua-Anderson) [inline cmd/apps.go]: This works, but it'd prefer to check for `fatal: remote <remote name> already exists.`, which is more explicit.

Comment (bacongobbler) [inline cmd/apps.go]: Good idea. This actually caught a bug in the unit tests I'm writing too. It was catching `[Error when running 'git remote add deis ssh://git@127-builder.0.0.1:2222/foo.git'
       		fatal: Not a git repository (or any of the parent directories): .git` which is of course an error we did not intend to catch.

File: cmd/apps_test.go [hunk 2]
```diff
@@ -27,3 +37,35 @@ func TestExpandUrl(t *testing.T) {
...
+    "owner": "jkirk",
+    "id": "foo",
+    "created": "2014-01-01T00:00:00UTC",
+    "updated": "2014-01-01T00:00:00UTC",
+    "uuid": "de1bf5b5-4a72-4f94-a10c-d2a3741cdf75"
+}`)
+	})
+
+	// create a remote first before running apps:create
+	assert.NoErr(t, git.Init())
```
Comment (Joshua-Anderson) [inline cmd/apps_test.go]: Can move the PWD to a temporary directory, so we're not creating lots of git remotes in the workflow-cli directory?

Comment (bacongobbler) [inline cmd/apps_test.go]: We run the tests in a container so it's wiped after each run. Are we to assume that these tests can be run out of the container?

Comment (Joshua-Anderson) [inline cmd/apps_test.go]: Yeah, I like running `go test` out of a container so I can get my <1 second test runs 😄

Comment (bacongobbler) [inline cmd/apps_test.go]: done.

File: cmd/apps.go [hunk 3]
```diff
@@ -50,7 +50,7 @@ func (d DeisCmd) AppCreate(id, buildpack, remote string, noRemote bool) error {
 
 	if !noRemote {
 		if err = git.CreateRemote(s.Client.ControllerURL.Host, remote, app.ID); err != nil {
-			if err.Error() == "exit status 128" {
+			if strings.Contains(err.Error(), fmt.Sprintf("fatal: remote %s already exists.", remote)) {
```
Comment (arschles) [inline cmd/apps.go]: @bacongobbler can you compare the error type here, instead of the error string?

File: pkg/git/git.go [hunk 4]
```diff
@@ -34,6 +34,17 @@ func CreateRemote(host, remote, appID string) error {
 	return nil
 }
 
+// Init creates a new git repository in the local directory.
+func Init() error {
+	cmd := []string{"git", "init"}
+	_, err := exec.Command(cmd[0], cmd[1]).Output()
+	if err != nil {
```
Comment (arschles) [inline pkg/git/git.go]: combine this and the above line: `if _, err := exec.Command(...); err != nil {`

File: cmd/apps.go (hunk 3 above)
Comment (bacongobbler) [inline cmd/apps.go]: Unfortunately all git errors fall under a generic `gitError` type so we cannot tell the difference between a `git remote already exists` error and a `not a git repository` error.

Comment (Joshua-Anderson) [inline cmd/apps.go]: In this case it isn't possible without larger rewrite. We could map STDERR parsing to errors in `git` package in the future, but I would like to rethink the git package in the future.

Comment (arschles) [inline cmd/apps.go]: ok, is there an issue for the rethink-git-package work?

Comment (Joshua-Anderson) [inline cmd/apps.go]: #196 is an error that is coming up due to differences in the git versions, but I'm kicking around the idea of using a full git go package to replace our homegrown client, but it's still in the feasibility stage right now.

**Pull request merged**